### PR TITLE
Document Song of Time Cutscenes (Test6)

### PR DESCRIPTION
**Pull request by engineer124**
z_en_test6.c is a cutscene actor that primarily handles the 3 variation of song of time cutscenes: 
1) The three-day reset Song of Time (Normal Song of Time)
2) The speed-up or slow-down of time (Inverted Song of Time)
3) The skip-ahead to the next half-day (Song of Double Time)

### CODE REVIEW

File: src/overlays/actors/ovl_En_Test6/z_en_test6.c [hunk 1]
```diff
@@ -821,24 +893,27 @@ void func_80A92188(EnTest6* this, PlayState* play) {
...
         }
-    } else if ((this->unk_27A < 96) && (this->unk_27A > 50) &&
+    } else if ((this->timer < 96) && (this->timer > 50) &&
                (CHECK_BTN_ALL(input->press.button, BTN_A) || CHECK_BTN_ALL(input->press.button, BTN_B))) {
-        this->unk_286 = 1;
-        this->unk_27A = 39;
+        this->screenFillAlpha = 1;
+        this->timer = 39;
         SEQCMD_STOP_SEQUENCE(SEQ_PLAYER_FANFARE, 20);
     }
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Test6/z_en_test6.c:913]: Huh, I didn't remember you could affect this effect by pressing A or B. What does this do?

File: src/overlays/actors/ovl_En_Test6/z_en_test6.c [hunk 2]
```diff
@@ -1094,228 +1173,245 @@ void func_80A92950(EnTest6* this, PlayState* play) {
...
-        Matrix_Translate(cos, phi_f24, sin, MTXMODE_NEW);
+    this->gfx = POLY_OPA_DISP;
+    clockPos.y = 0.0f;
+
+    clock1Yaw = this->clockAngle;
+    clock2Yaw = clock1Yaw + 0x4E20 + (s32)(0x2EE0 * Math_SinS(play->state.frames));
+    angle = (play->state.frames & 0x3C) * 1024;
+    angle *= this->clockSpeed / 200.0f;
+    this->clockAngle += (s16)this->clockSpeed;
+    this->clockRingRotZ = (s16)((this->clockSpeed / 200.0f) * 256.0f);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Test6/z_en_test6.c:1218]: Those are some whacky maths monkaS

**Review by AngheloAlf — COMMENTED** [review]
good job on this :3

File: src/overlays/actors/ovl_En_Test6/z_en_test6.c (hunk 1 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Test6/z_en_test6.c:913]: Mashing A/B actually shortens these cutscenes by a little bit. Specifically, the cutscene starts at a timer with a high number, then it counts down to 0 to finish. If the timer is between 96 -> 50, then pressing A/B will skip parts of the cutscene directly to timer 39 and skip the steps in between

File: src/overlays/actors/ovl_En_Test6/z_en_test6.c [hunk 3]
```diff
@@ -650,63 +711,71 @@ void func_80A92118(EnTest6* this, PlayState* play) {
...
-        this->unk_160 += 0.2f;
-        func_80A90C54(play, this->unk_160);
-    } else if (this->unk_27A > 90) {
-        this->unk_160 -= 0.05f;
-        func_80A90C54(play, this->unk_160);
-    } else if (this->unk_27A == 90) {
-        this->unk_160 = 0.0f;
-        func_80A90D20(play);
+    if (this->timer > 115) {
+        this->doubleSoTEnvLerp += 0.2f;
```
[the remark this answers is not in the document]
Comment (engineer124) [reply src/overlays/actors/ovl_En_Test6/z_en_test6.c:731]: I wasn't able to figure out the fractions `0.9333333f` or `0.06666666f` although I didn't try too many combinations.

File: src/overlays/actors/ovl_En_Test6/z_en_test6.c (hunk 2 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Test6/z_en_test6.c:1218]: Oh yeah, I worked out that the `& 0x3C` ensures the `angle` updates only once every 4 frames. I made that comment elsewhere but will add that in here

File: src/overlays/actors/ovl_En_Test6/z_en_test6.c [hunk 4]
```diff
@@ -848,166 +923,170 @@ void EnTest6_Update(Actor* thisx, PlayState* play) {
...
+            case SOTCS_CUEID_NONE:
             default:
-                this->unk_276 = 99;
+                this->drawType = SOTCS_DRAW_TYPE_NONE;
                 return;
 
-            case 9:
+            case SOTCS_CUEID_DOUBLE_END:
                 Play_SetRespawnData(&play->state, 1, ((void)0, gSaveContext.save.entrance & 0xFFFF), player->unk_3CE,
                                     PLAYER_PARAMS(0xFF, PLAYER_INITMODE_B), &player->unk_3C0, player->unk_3CC);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Test6/z_en_test6.c:1082]:
```suggestion
                Play_SetRespawnData(&play->state, RESPAWN_MODE_RETURN, ((void)0, gSaveContext.save.entrance), player->unk_3CE,
                                    PLAYER_PARAMS(0xFF, PLAYER_INITMODE_B), &player->unk_3C0, player->unk_3CC);
```
Locally the `& 0xFFFF` doesn't seem to be needed, and can use the `RESPAWN_MODE` enum.

File: src/overlays/actors/ovl_En_Test6/z_en_test6.c [hunk 5]
```diff
@@ -1016,74 +1095,74 @@ void func_80A92950(EnTest6* this, PlayState* play) {
...
-                this->unk_276 = 99;
+                this->drawType = SOTCS_DRAW_TYPE_NONE;
                 return;
 
-            case 9:
+            case SOTCS_CUEID_DOUBLE_END:
                 if (gSaveContext.save.time > CLOCK_TIME(12, 0)) {
                     Play_SetRespawnData(&play->state, 1, ((void)0, gSaveContext.save.entrance & 0xFFFF),
                                         player->unk_3CE, PLAYER_PARAMS(0xFF, PLAYER_INITMODE_B), &player->unk_3C0,
                                         player->unk_3CC);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Test6/z_en_test6.c:1164]:
```suggestion
                    Play_SetRespawnData(&play->state, RESPAWN_MODE_RETURN, ((void)0, gSaveContext.save.entrance),
                                        player->unk_3CE, PLAYER_PARAMS(0xFF, PLAYER_INITMODE_B), &player->unk_3C0,
                                        player->unk_3CC);
```
Same here.

**Review by hensldm — COMMENTED** [review]
Just 1 thing, that occurs in 2 places.

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]

**Pull request merged**